### ISSUE TITLE
Support nixos

### ISSUE THREAD

**Issue by mechanicstriker** [devcontainers/images#717]
🙄🙄

**Comment by samruddhikhandale** [comment]
Hi 👋 

Thanks for your interest, we appreciate it! ✨ 

Currently, I am not aware of any plans to support NixOS (out of the box). Let's keep this issue open and get more feedback and interest from the community. If there's decent amount of interest, we can reiterate on our plans. @mechanicstriker How does that sound? 

Also, we do support the [Nix Feature](https://github.com/devcontainers/features/tree/main/src/nix), @mechanicstriker wondering, have you looked at that?

**Comment by joshspicer** [comment]
Would you mind elaborating on this request?  This repo contains source for a set of docker images built for several platforms.

**Comment by mechanicstriker** [comment]
Greetings,

Thank you for the response and consideration. I'd like to provide a more detailed rationale for the inclusion of a native NixOS image support.

The primary objective behind this request is to facilitate a more streamlined development environment, particularly for developers working with tools and languages like Rust in conjunction with the Nix package manager. At present, achieving an optimal setup necessitates the use of additional VSCode extensions, such as `nix-env-selector`, which exposes the `nix-shell` environment to VSCode. Were there a native NixOS image available, one could straightforwardly integrate the Rust toolchain as a system package, eliminating the current complexities and workarounds.

Moreover, when leveraging the Nix feature atop another Docker image, one encounters pre-installed packages that could potentially result in conflicts. From my perspective, a dedicated NixOS image would provide a cleaner, conflict-free solution, ensuring a more robust and predictable development experience.

I firmly believe that such an enhancement would not only benefit my workflow but would also be of value to the broader community who might be grappling with similar challenges. I appreciate your openness to community feedback and hope this elaboration clarifies the utility and significance of the proposed native NixOS image.

Warm regards,
Adrian

**Comment by jkeech** [comment]
It sounds to me like this is a request for a `base-nixos` image, similar to the other `base-*` images (alpine, debian, ubuntu). It doesn't mean we would need to provide language-specific images on top of the nixos base (although we could in some cases), but at least it would give folks a nice starting point to build up a custom image of their choosing. @mechanicstriker does that align with how you are thinking about it?

**Comment by mechanicstriker** [comment]
Yes, you've understood correctly. A `base-nixos` image, similar to the existing base images, would be ideal. This would serve as a foundational platform for developers to build upon.
